cupy/linalg/norms.py

```diff
@@ -253,22 +253,25 @@ def _slogdet_one(a):
     getrf(handle, m, m, a_copy.data.ptr, m, workspace.data.ptr,
           ipiv.data.ptr, dev_info.data.ptr)
 
-    try:
-        cupy.linalg.util._check_cusolver_dev_info_if_synchronization_allowed(
-            getrf, dev_info)
-
-        diag = cupy.diag(a_copy)
-        # ipiv is 1-origin
-        non_zero = (cupy.count_nonzero(ipiv != cupy.arange(1, m + 1)) +
-                    cupy.count_nonzero(diag < 0))
-        # Note: sign == -1 ** (non_zero % 2)
-        sign = (non_zero % 2) * -2 + 1
-        logdet = cupy.log(abs(diag)).sum()
-    except linalg.LinAlgError:
-        sign = cupy.array(0.0, dtype=dtype)
-        logdet = cupy.array(float('-inf'), dtype)
-
-    return sign, logdet
+    # dev_info < 0 means illegal value
+    dev_info_err = cupy.fmin(dev_info, 0)
```

Comment:
This should be done after `_check_cusolver_dev_info_if_synchronization_allowed` to raise LinAlgError if `errstate(linalg='raise')`.

Reply:
Thanks.  Stopped raising `LinAlgError` for bugs in CuPy. (64a6b7e0df9c191a3194e2786156f46b6de3bfa5)

Reply:
Thanks!

Reply:
Thank you for pointing out & fixing.
"Illegal value" could be misunderstood as invalid values of arrays (e.g. `nan` in `a`).
How about different wording like "Illegal usage that should never happen"?

Reply:
Added the explanation.  I'd like to keep "illegal value" because the error message "parameter number %d had an illegal value" looks common to me in matrix libraries.

Reply:
OK, thank you for the fix!